ocaml/typing/typedecl.ml

```diff
@@ -626,8 +626,8 @@ let transl_declaration env sdecl (id, uid) =
   let params = List.map (fun (cty, _) -> cty.ctyp_type) tparams in
   let cstrs = List.map
     (fun (sty, sty', loc) ->
-      transl_simple_type env ~closed:false Mode.Alloc.Const.legacy sty,
-      transl_simple_type env ~closed:false Mode.Alloc.Const.legacy sty', loc)
+      transl_simple_type ~new_var_jkind:Any env ~closed:false Mode.Alloc.Const.legacy sty,
```

Comment:
Definitely worth a comment explaining the choices here.

Even better, put an explanation on `transl_simple_type` that walks the reader through some of our design process; a reader who is wondering why a particular choice is made will surely check for comments on `transl_simple_type`, and those comments will explain the situation.